### PR TITLE
Add engineReinitMarkers()

### PR DESCRIPTION
**Pull request by TFP-dev**
This function will reinit 3D markers. This can be useful if you want to replace marker 3D model. Fixes closed issue #2213.

**API:**
`engineReinitMarkers()`

**Test resource:** 
This resource replaces "arrow" marker and adds command `reinitMarkers`.
[marker_test_resource.zip](https://github.com/multitheftauto/mtasa-blue/files/9832216/marker_test_resource.zip)

**Screenshot:**
comp

### CODE REVIEW

**Comment by CrosRoad95** [comment]
". This can be useful if you want to replace marker 3D model."
so, make if you replace 3d model of marker -> call "engineReinitMarkers", we don't use "reinitVehicles", "reinitObjects" ect after element will be replaced

**Comment by TFP-dev** [comment]
> ". This can be useful if you want to replace marker 3D model." so, make if you replace 3d model of marker -> call "engineReinitMarkers", we don't use "reinitVehicles", "reinitObjects" ect after element will be replaced

But we use `engineRestreamWorld`. Currently servers can use fact that you can't replace marker model so `engineReinitMarkers` is backwards compatible.

**Comment by CrosRoad95** [comment]
you actually fixing a bug, everyone will happy if suddenly markers will work as intended

**Comment by TFP-dev** [comment]
> you actually fixing a bug, everyone will happy if suddenly markers will work as intended

Check issue #2213

**Comment by lopezloo** [comment]
This issue applies to `cylinder` markers too, right?

**Review by sbx320 — COMMENTED** [review]
Wouldn't it be more useful to apply this whenever we replac a marker model? That way scripters wouldn't need to work around it.

**Comment by TFP-dev** [comment]
> Wouldn't it be more useful to apply this whenever we replac a marker model? That way scripters wouldn't need to work around it.

Currently you can use fact that you can't replace markers, for example you can replace id 1559 with the building model and use it instead of allocating new model id with `engineRequestModel` and markers will stay intact.

File: Client/game_sa/C3DMarkersSA.cpp [hunk 1]
```diff
@@ -122,3 +122,15 @@ C3DMarker* C3DMarkersSA::FindMarker(DWORD Identifier)
     }
     return NULL;
 }
+
+void C3DMarkersSA::ReinitMarkers()
+{
+    typedef int(__cdecl * Function_ShutdownMarkers)();
```
Comment (sbx320) [inline Client/game_sa/C3DMarkersSA.cpp]: Should be using rather than typedef

File: Client/mods/deathmatch/logic/luadefs/CLuaEngineDefs.cpp [hunk 2]
```diff
@@ -2058,3 +2059,9 @@ bool CLuaEngineDefs::EngineRestreamWorld(lua_State* const luaVM)
     g_pClientGame->RestreamWorld();
     return true;
 }
+
+bool CLuaEngineDefs::EngineReinitMarkers(lua_State* const luaVM)
```
Comment (sbx320) [inline Client/mods/deathmatch/logic/luadefs/CLuaEngineDefs.cpp]: No need to return a bool here, if it's always `true`. Also if the lua_State* arg isn't used, it doesn't need to be specified. 

The argparse can handle `void [func]()`.

**Review by sbx320 — DISMISSED** [review]
Some minor comments regarding argparser usage

**Comment by lopezloo** [comment]
> Wouldn't it be more useful to apply this whenever we replac a marker model? That way scripters wouldn't need to work around it.

Maybe that could be implemented and documented as 1.6 backwards incompatible change.

**Comment by botder** [comment]
Superseded by pull request #3059.

**Pull request closed without merging**